java-extensions/hudi-reader/src/main/java/com/starrocks/hudi/reader/HudiSliceScanner.java

```diff
@@ -88,68 +94,91 @@ public HudiSliceScanner(int fetchSize, Map<String, String> params) {
         this.classLoader = this.getClass().getClassLoader();
     }
 
-    @Override
-    public void open() throws IOException {
-        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
-            Properties properties = new Properties();
-            Configuration conf = new Configuration();
-            conf.setBoolean("dfs.client.use.legacy.blockreader", false);
-            JobConf jobConf = new JobConf(conf);
-            jobConf.setBoolean("hive.io.file.read.all.columns", false);
-            String[] hiveColumnNames = this.hiveColumnNames.split(",");
-            HashMap<String, Integer> hiveColumnNameToIndex = new HashMap<>();
-            HashMap<String, String> hiveColumnNameToType = new HashMap<>();
-            for (int i = 0; i < hiveColumnNames.length; i++) {
-                hiveColumnNameToIndex.put(hiveColumnNames[i], i);
-                hiveColumnNameToType.put(hiveColumnNames[i], hiveColumnTypes[i]);
-            }
+    private JobConf makeJobConf(Properties properties) {
+        Configuration conf = new Configuration();
+        conf.setBoolean("dfs.client.use.legacy.blockreader", false);
+        JobConf jobConf = new JobConf(conf);
+        jobConf.setBoolean("hive.io.file.read.all.columns", false);
+        properties.stringPropertyNames().forEach(name -> jobConf.set(name, properties.getProperty(name)));
+        return jobConf;
+    }
 
-            String[] requiredTypes = new String[requiredFields.length];
-            StringBuilder columnIdBuilder = new StringBuilder();
-            boolean isFirst = true;
-            for (int i = 0; i < requiredFields.length; i++) {
-                if (!isFirst) {
-                    columnIdBuilder.append(",");
-                }
-                columnIdBuilder.append(hiveColumnNameToIndex.get(requiredFields[i]));
-                String typeStr = hiveColumnNameToType.get(requiredFields[i]);
-                // convert decimal(x,y) to decimal
-                if (typeStr.startsWith("decimal")) {
-                    typeStr = "decimal";
-                }
-                requiredTypes[i] = typeStr;
-                isFirst = false;
+    private void parseRequiredTypes() {
+        String[] hiveColumnNames = this.hiveColumnNames.split(",");
+        HashMap<String, Integer> hiveColumnNameToIndex = new HashMap<>();
+        HashMap<String, String> hiveColumnNameToType = new HashMap<>();
+        for (int i = 0; i < hiveColumnNames.length; i++) {
+            hiveColumnNameToIndex.put(hiveColumnNames[i], i);
+            hiveColumnNameToType.put(hiveColumnNames[i], hiveColumnTypes[i]);
+        }
+
+        requiredTypes = new ColumnType[requiredFields.length];
+        requiredColumnIds = new Integer[requiredFields.length];
```

Comment:
```
private int[] requiredColumnIds;
..
requiredColumnIds = new int[requiredFields.length];
```

use primitive array to avoid autoboxing and unboxing